mtp_api/apps/performance/models.py

```diff
@@ -1,13 +1,76 @@
 import datetime
 import itertools
 
+from django.conf import settings
 from django.db import connection, models
 from django.db.models.expressions import RawSQL
+from django.utils import timezone
 from django.utils.translation import gettext_lazy as _
 
 from core import dictfetchall, mean
 
 
+class DigitalTakeupManager(models.Manager):
+    def digital_takeup_per_month(self, since=None, exclude_private_estate=False):
+        """
+        Per-month digital take-up averages
+        """
+        if since is None:
+            today = timezone.localdate()
+            since = max(DigitalTakeup.reports_start, today.replace(year=today.year - 2, month=1, day=1))
+
+        if exclude_private_estate:
+            included_prisons_sql = """
+                included_prisons AS (
+                    SELECT prison_prison.nomis_id as nomis_id
+                    FROM prison_prison
+                    WHERE prison_prison.private_estate IS false
+                ),
+            """
+            credit_count_join_sql = """
+                JOIN included_prisons ON included_prisons.nomis_id = credit_credit.prison_id
+            """
+            average_takeup_join_sql = """
+                JOIN included_prisons ON included_prisons.nomis_id = performance_digitaltakeup.prison_id
+            """
+        else:
+            included_prisons_sql = ''
+            credit_count_join_sql = ''
+            average_takeup_join_sql = ''
+
+        sql = f"""
+        WITH
+            {included_prisons_sql}
+            credit_count AS (
+                SELECT
+                    date_trunc('month', received_at AT TIME ZONE '{settings.TIME_ZONE}')::timestamp WITH TIME ZONE
```

Comment:
that sounds interesting… could be a good thing to try. i worry that it might affect loads of different parts of the code though. also, TIME_ZONE wouldn't be anything other than Europe/London